src/nanotron/trainer.py

```diff
@@ -333,9 +333,16 @@ def clear_dataloader_from_memory(dataloader: DataLoader, stage_name: str):
                     if isinstance(prev_dataloader, DataLoader):
                         # NOTE: we don't need to clear dummy data generator from memory
                         clear_dataloader_from_memory(prev_dataloader, stage_name=stage.name)
+                
+                # some datasets in brrr doesn't have hf_dataset_or_datasets attribute, like TokenizedBytesDatasetArgs
+                if hasattr(stage.data.dataset, 'hf_dataset_or_datasets'):
+                    dataset_info = f"[Training Stage: {stage.name}] Switching to a new dataset {stage.data.dataset.hf_dataset_or_datasets}"
```

Comment:
if `hf_dataset_or_datasets` attr doesnt exist, would that create an issue for later?

Reply:
No, this is only to print some information.

Reply:
I have trained a model with TokenizedBytesDatasetArgs which doesn't have hf_dataset_or_datasets. Before added the check, it will generate an error for it